redaxo/src/core/lib/sql/table.php

```diff
@@ -977,4 +977,41 @@ private function resetModified()
             $this->foreignKeysExisting[$foreignKey->getName()] = $foreignKey->getName();
         }
     }
+
+
+
+    /**
+     * Copy the table structure to another table.
+     * @param string $destinationTable
+     *
+     * @throws rex_exception
+     */
+    public function copyTo($destinationTable)
+    {
+        if ($this->new) {
+            throw new rex_exception(sprintf('Table "%s" does not exist.', $this->originalName));
+        }
+
+        if (self::get($destinationTable)->exists()) {
+            throw new rex_exception(sprintf('Table "%s" already exists.', $destinationTable));
+        }
+
+        $query = 'CREATE TABLE '.$this->sql->escapeIdentifier($destinationTable).' LIKE '.$this->sql->escapeIdentifier($this->originalName);
+        $this->sql->setQuery($query);
+    }
+
+
+    /**
+     * Copy the table structure and its data to another table.
+     * @param string $destinationTable
+     *
+     * @throws rex_exception
+     */
+    public function copyWithDataTo($destinationTable)
```

Comment:
Bisher arbeitet diese Klasse nur auf Strukturebene, nicht mit den Daten.
Da bin ich mir auch noch nicht ganz sicher, ob man die Trennung lieber lässt, oder eben nicht (und ggf auch noch weitere Datenorientierte Methoden reinnimmt).

Reply:
Da Daten zu einer Tabelle gehören, fände ich es schon gut, wenn wir dies mit aufnehmen.

Reply:
(ich denke gregor meinte, ob man diese methode hier lässt, oder in einer neuen klasse)